crates/runtime/src/embeddings/index/query_table.rs

```diff
@@ -297,7 +299,7 @@ impl VectorQueryTableProvider {
         ];
 
         query_table_projection_exprs.extend(metadata_columns_to_exprs(
-            self.vector_index.metadata_columns(),
+            &self.vector_index.metadata_columns(),
```

Comment:
Why did you make `metadata_columns()` return an owned clone when you just reference it?